### ISSUE TITLE
Master Linting and Test Failures

### ISSUE THREAD

**Issue by grippy** [redis/redis-py#2194]
Thanks for wanting to report an issue you've found in redis-py. Please delete this text and fill in the template below.  
It is of course not always possible to reduce your code to a small test case, but it's highly appreciated to have as much data as possible. Thank you!

**Version**: What redis-py and what redis version is the issue happening on?

master branch starting at commit: `a53bc0ca01211e5e14a468077a9d1974dbba6054`

**Platform**: What platform / version? (For example Python 3.5.1 on Windows 7 / Ubuntu 15.10 / Azure)

**Description**: Description of your issue, stack traces from errors and code that reproduces the issue

Master was all green before this commit: https://github.com/redis/redis-py/commit/a53bc0ca01211e5e14a468077a9d1974dbba6054

I have a branch to which shows some of the failures (and fixes the linting failure) here:
https://github.com/redis/redis-py/runs/6516808625?check_suite_focus=true

I'm seeing additional failures when I run the testsuite locally, too.

**Comment by grippy** [comment]
```
___________________ test_clear_dollar[single-python-parser] ____________________

modclient = Redis<ConnectionPool<Connection<host=localhost,port=36379,db=0>>>

    @pytest.mark.redismod
    async def test_clear_dollar(modclient: redis.Redis):
    
        await modclient.json().set(
            "doc1",
            "$",
            {
                "nested1": {"a": {"foo": 10, "bar": 20}},
                "a": ["foo"],
                "nested2": {"a": "claro"},
                "nested3": {"a": {"baz": 50}},
            },
        )
        # Test multi
>       assert await modclient.json().clear("doc1", "$..a") == 4
E       assert 3 == 4
````

Debugging this, it looks like the redis JSON module isn't properly clearing the `a` property values. It's skipping `nested2.a`:

```
127.0.0.1:6379> JSON.SET "doc1" "$" '{"nested1": {"a": {"foo": 10, "bar": 20}}, "a": ["foo"], "nested2": {"a": "claro"}, "nested3": {"a": {"baz": 50}}}'
OK

127.0.0.1:6379> JSON.GET "doc1"
"{\"nested1\":{\"a\":{\"foo\":10,\"bar\":20}},\"a\":[\"foo\"],\"nested2\":{\"a\":\"claro\"},\"nested3\":{\"a\":{\"baz\":50}}}"

127.0.0.1:6379> JSON.CLEAR "doc1" "$..a"
(integer) 3

127.0.0.1:6379> JSON.GET "doc1"
"{\"nested1\":{\"a\":{}},\"a\":[],\"nested2\":{\"a\":\"claro\"},\"nested3\":{\"a\":{}}}"
```

This was introduced by this change:

https://github.com/RedisJSON/RedisJSON/pull/721/files#diff-1ec7eb96047050ed1b91b693840c276d1cd5ea765d41f7aa6b47f8e68d89dd74

And the test linked above doesn't expect "$.nested2.a" to change.

Going to change the failing test to match the new expected output.

**Comment by chayim** [comment]
Thanks @grippy . You're right - upstream there were changes to RedisJSON that necessitate fixes. @oshadmi can potentially comment on those even.

**Comment by oshadmi** [comment]
@grippy The change indeed occurred in the PR you mentioned,
Which no longer clears String and Bool scalar values.
This change is no longer expecting `"$.nested2.a"` to be cleared, and it keeps its String value (`"claro"`)
https://github.com/RedisJSON/RedisJSON/pull/721/files#diff-1ec7eb96047050ed1b91b693840c276d1cd5ea765d41f7aa6b47f8e68d89dd74L752

**Comment by chayim** [comment]
Closed by PR #2191
